doc/configuration_system.md

```diff
@@ -0,0 +1,172 @@
+* mysql_query_rules -- the list of rules for routing traffic to the different backend servers. These rules can also cause a rewrite of the query, or caching of the result
+* global_variables -- the list of global variables used throughout the proxy that can be tweaked at runtime. Examples of global variables:
+```
+mysql> select * from global_variables limit 3;
++----------------------------------+----------------+
+| variable_name                    | variable_value |
++----------------------------------+----------------+
+| mysql-connect_retries_on_failure | 5              |
+| mysql-connect_retries_delay      | 1              |
+| mysql-connect_timeout_server_max | 10000          |
++----------------------------------+----------------+
+```
+* mysql_collations -- the list of MySQL collations available for the proxy to work with. See [this](http://stackoverflow.com/questions/341273/what-does-character-set-and-collation-mean-exactly) StackOverflow answer for the difference between a collation and a charset.
+* [only available in debug builds] debug_levels -- the list of types of debug statements that ProxySQL emits together with their verbosity levels. This allows us to easily configure at runtime what kind of statements we have in the log in order to debug different problems
+
+__DISK__ and __CONFIG FILE__
+
+__DISK__ represents an on-disk SQLite3 database, with the default location at
+`/tmp/proxysql.db`. Across restarts, the in-memory configs that were not
+persisted will be gone. __CONFIG__ file is the classical config file, and we'll
+see the relationship between it and the other configuration layers in the next
+section.
+
+In the following sections, we'll describe the lifecycle of each of these layers
+for the basic operations that the daemon goes through: starting up for the
+first time, starting up, restarting, shutting down, etc.
+
+# Startup
+
+At a normal start-up, ProxySQL initializes its configuration from the persisted
+in-memory database. This is found in the "datadir" (usually the current working directory in which ProxySQL is ran). The filename for the database is named "proxysql.db". So, disk gets loaded into memory and then propagated towards the
+runtime configuration.
+
+# Initial startup (or --initial flag)
+
+At the initial start-up (or start-up done with --initial flag, which resets the
+database by renaming the old one), the major difference from normal start-up is that the memory and runtime configuration gets populated from the config file.
+
+After this is done, the configuration is also persisted to the disk database,
+which will be used for the next restarts.
+
+# Soft restart (via angel process)
+
+# Hard restart
+
+# Modifying config at runtime
+
+Modifying the config at runtime is done through the MySQL admin port of ProxySQL. After connecting to it, we are presented with a MySQL-compatible interface for querying several ProxySQL-related tables:
+```mysql
+mysql> show tables;
```

Comment:
We need to describe these tables in more details, what their fields represent, and which fields are there but currently not supported (placeholders)